motionplan/nearestNeighbor.go

```diff
@@ -138,43 +124,29 @@ func (nm *neighborManager) nnWorker(ctx context.Context, planOpts *plannerOption
 	var best node
 	bestDist := math.Inf(1)
 
-	for {
-		select {
-		case <-ctx.Done():
-			nm.neighbors <- nil
-			return
-		default:
+	for candidate := range nm.nnKeys {
+		seg := &Segment{
+			StartConfiguration: nm.seedPos.Q(),
+			EndConfiguration:   candidate.Q(),
+		}
+		if pose := nm.seedPos.Pose(); pose != nil {
+			seg.StartPosition = pose
+		}
+		if pose := candidate.Pose(); pose != nil {
+			seg.EndPosition = pose
+		}
+		dist := planOpts.DistanceFunc(seg)
+		if dist < bestDist {
+			bestDist = dist
+			best = candidate
 		}
 
 		select {
-		case k := <-nm.nnKeys:
-			if k != nil {
-				nm.nnLock.RLock()
-				seg := &Segment{
-					StartConfiguration: nm.seedPos.Q(),
-					EndConfiguration:   k.Q(),
-				}
-				if pose := nm.seedPos.Pose(); pose != nil {
-					seg.StartPosition = pose
-				}
-				if pose := k.Pose(); pose != nil {
-					seg.EndPosition = pose
-				}
-				dist := planOpts.DistanceFunc(seg)
-				nm.nnLock.RUnlock()
-				if dist < bestDist {
-					bestDist = dist
-					best = k
-				}
-			}
+		case <-ctx.Done():
+			break
 		default:
-			nm.nnLock.RLock()
-			if nm.ready {
```

Comment:
We're able to remove the `ready` field because we now learn all inputs were consumed by seeing the channel was closed.